packages/inventory/src/Stocktake/DetailView/DetailView.tsx

```diff
@@ -1,173 +1,63 @@
-import React, { FC } from 'react';
+import React, { FC, useState } from 'react';
 import {
-  useParams,
   TableProvider,
   createTableStore,
-  useOmSupplyApi,
-  useDocument,
-  DataTable,
-  usePagination,
-  useTranslation,
-  Box,
-  Column,
-  useDialog,
-  DialogButton,
-  useColumns,
-  ColumnAlign,
-  GenericColumnKey,
+  Item,
+  ModalMode,
 } from '@openmsupply-client/common';
-import { reducer, StocktakeActionCreator } from './reducer';
-import { getStocktakeDetailViewApi } from './api';
+
 import { Toolbar } from './Toolbar';
 import { Footer } from './Footer';
 import { AppBarButtons } from './AppBarButtons';
 import { SidePanel } from './SidePanel';
-import { StocktakeItem } from '../../types';
-import { StocktakeLineEdit } from './modal/StocktakeLineEdit/StocktakeLineEdit';
-
-const useDraftStocktake = () => {
-  const { id } = useParams();
-  const { api } = useOmSupplyApi();
-
-  const { draft, save, dispatch, state } = useDocument(
-    ['stocktake', id],
-    reducer,
-    getStocktakeDetailViewApi(api)
-  );
-
-  const onChangeSortBy = (column: Column<StocktakeItem>) => {
-    dispatch(StocktakeActionCreator.sortBy(column));
-  };
-
-  return { draft, save, dispatch, onChangeSortBy, sortBy: state.sortBy };
-};
-
-const Expand: FC<{ rowData: StocktakeItem }> = ({ rowData }) => {
-  return (
-    <Box p={1} height={300} style={{ overflow: 'scroll' }}>
-      <Box
-        flex={1}
-        display="flex"
-        height="100%"
-        borderRadius={4}
-        bgcolor="#c7c9d933"
-      >
-        <span style={{ whiteSpace: 'pre-wrap' }}>
-          {JSON.stringify(rowData, null, 2)}
-        </span>
-      </Box>
-    </Box>
-  );
-};
-
-export enum ModalMode {
-  Create,
-  Update,
-}
+import { StocktakeLine, StocktakeSummaryItem } from '../../types';
+import { StocktakeLineEdit } from './modal/StocktakeLineEdit';
+import { ContentArea } from './ContentArea';
+
+export const toItem = (line: StocktakeLine | StocktakeSummaryItem): Item => ({
+  id: 'lines' in line ? line.lines[0].itemId : line.itemId,
+  name: 'lines' in line ? line.lines[0].itemName : line.itemName,
+  code: 'lines' in line ? line.lines[0].itemCode : line.itemCode,
+  isVisible: true,
+  availableBatches: [],
+  availableQuantity: 0,
+  unitName: 'bottle',
+});
 
 export const DetailView: FC = () => {
-  const { hideDialog, showDialog, Modal } = useDialog({
-    onClose: () => setModalState({ item: null, mode: ModalMode.Create }),
-  });
-  const { draft, onChangeSortBy, sortBy } = useDraftStocktake();
-
-  const onChangeItem = (item: StocktakeItem | null) => {
-    setModalState(state => ({ ...state, item }));
-  };
+  const [modalState, setModalState] = useState<{
+    item: Item | null;
+    mode: ModalMode;
+    isOpen: boolean;
+  }>({ item: null, mode: ModalMode.Create, isOpen: false });
```

Comment:
TODO:  Made a `useEditModal` hook in `location/ListView` which could be utilised for all of the edit modals